test/cases/channel/deposit.test.ts

```diff
@@ -0,0 +1,110 @@
+import setupTestEnvironment from '../../../src';
+import p2p from '../../../topologies/p2p.json';
+import { tokenAddresses, toWei } from '../../../src/util/token';
+import { LuminoTestEnvironment } from '../../../src/types/lumino-test-environment';
+import { Dictionary } from '../../../src/util/collection';
+import { LuminoNode } from '../../../src/types/node';
+import { verifyChannel } from '../../utils';
+import {
+    SETUP_TIMEOUT,
+    TEARDOWN_TIMEOUT,
+    ChannelState,
+    State,
+    TEST_TIMEOUT,
+} from '../../common';
+
+interface depositParams {
+    token: string;
+    amount: number;
+    partner: string;
+}
+
+describe('channel deposit', () => {
+    let nodes: Dictionary<LuminoNode>;
+    let env: LuminoTestEnvironment;
+
+    beforeAll(async () => {
+        env = await setupTestEnvironment(p2p);
+        nodes = env.nodes as Dictionary<LuminoNode>;
+    }, SETUP_TIMEOUT);
+
+    afterAll(async () => {
+        await env.stop();
+    }, TEARDOWN_TIMEOUT);
```

Comment:
I agree but let's not change all this into a black box code please :) we should see the limit between improve the test code and write a jest test, the hooks code probably will be pretty similar on a lot of tests but we need to keep in mind that probably for some test we will need to make changes.

Reply:
> I agree but let's not change all this into a black box code please :) we should see the limit between improve the test code and write a jest test, the hooks code probably will be pretty similar on a lot of tests but we need to keep in mind that probably for some test we will need to make changes.

sure. let's leave it as it is for now. i don't think extracting this would make it "black box code", but i see your point